src/session/content/message_row/reply.rs

```diff
@@ -0,0 +1,222 @@
+                "reply-id" => obj.imp().reply_id.get().to_value(),
+                _ => unimplemented!(),
+            }
+        }
+
+        fn set_property(&self, _id: usize, value: &glib::Value, _pspec: &glib::ParamSpec) {
+            let obj = self.obj();
+            match _pspec.name() {
+                "chat" => obj.set_chat(Some(value.get().unwrap())),
+                "reply-id" => obj.set_reply_id(value.get().unwrap()),
+                _ => unimplemented!(),
+            }
+        }
+        fn constructed(&self) {
+            self.is_loading.set(true);
+            self.message_label
+                .set_label(&gettextrs::gettext("Loading ..."));
+        }
+
+        fn dispose(&self) {
+            while let Some(child) = self.obj().first_child() {
+                child.unparent();
+            }
+        }
+    }
+
+    impl WidgetImpl for MessageReply {}
+}
+
+glib::wrapper! {
+    pub(crate) struct MessageReply(ObjectSubclass<imp::MessageReply>)
+        @extends gtk::Widget;
+}
+
+impl MessageReply {
+    pub(crate) fn new(chat: Chat, reply_id: i64) -> Self {
+        glib::Object::builder()
+            .property("chat", chat)
+            .property("reply-id", reply_id)
+            .build()
+    }
+
+    pub(crate) fn set_chat(&self, chat: Option<Chat>) {
+        if self.imp().chat.borrow().clone() == chat {
+            return;
+        }
+
+        self.imp().chat.replace(chat);
+        self.notify("chat");
+    }
+
+    pub(crate) fn set_reply_id(&self, reply_id: i64) {
+        let imp = self.imp();
+
+        if imp.reply_id.get() == reply_id {
+            return;
+        }
+
+        imp.reply_id.set(reply_id);
+        self.notify("reply-id");
+    }
+
+    pub(crate) fn load_message(&self) {
+        let imp = self.imp();
+
+        let reply_id = imp.reply_id.get();
+        let chat = imp.chat.borrow().clone().unwrap();
+
+        if let Some(message) = chat.message(reply_id) {
+            self.update_from_message(&message);
+        } else {
+            spawn(clone!(@weak self as obj => async move {
+                let chat = obj.imp().chat.borrow().clone().unwrap();
+                if let Some(message) = chat.fetch_chat_message(reply_id).await {
+                    obj.update_from_message(&message);
+                } else {
+                    // Message doesn't exist, so we should remove "Loading..." caption
+                    // TODO: Impelent it properly using signals
+                    obj.imp().message_label.set_label("Deleted message");
+                }
+            }));
+        }
+
+        imp.is_loading.set(false);
+    }
+
+    pub(crate) fn set_outside_the_bubble(&self, outside: bool) {
+        if outside {
+            self.imp()
+                .message_label
+                .set_max_width_chars(STICKER_CHAR_LIMIT);
+        } else {
+            self.imp().message_label.set_max_width_chars(-1);
+        }
+    }
```

Comment:
@melix99 what do you think about this function? It's used to limit chars count of the replies in MessageSticker so it doesn't look like that
![Screenshot from 2023-03-06 18-20-53](https://user-images.githubusercontent.com/110191114/223184742-c0892837-181c-4acc-bdc8-c3223e719565.png)

Reply:
maybe I was thinking about changing the name to  `set_should_limit_chars_count`, but I feel like it's to long. I can also pass widget type to the contructor and set this property internally

Reply:
I agree that we need to limit the width, but I'm not sure about the implementation because I think this should be decided by the users of `MessageReply` (so `MessageBubble` and `MessageSticker`) rather than implementing it internally.

I think this can be done via CSS using `min-width` and then set `halign=end` in `MessageReply`.

Reply:
So something like:

```css

messagesticker messagreply {
  min-width: 200px;
}

```

Reply:
I'm afraid setting `max-width` doesn't work in gtk

Reply:
I know, but I said to use `min-width` :)

Reply:
I don't understand how `min-width` would help.

Reply:
I said to set `halign=end` to `MessageReply`, this way the widget is forced to shrink horizontally. Then you set the minimum width via `min-width` and the widget will have the wanted minimum width.

Reply:
> and the widget will have the wanted minimum width

We want to set max-width not min-width (image)
![Screenshot from 2023-03-06 18-59-26](https://user-images.githubusercontent.com/110191114/223192917-dcafae90-4050-40ae-a275-d31fe8bb74c5.png)

Reply:
Well, maybe min-width too, but not 200px. I think 100px should be fine

Reply:
> this way the widget is forced to shrink horizontally

Apparently this only works for `GtkInscription`. But we want `GtkLabel` for other reasons, so nevermind.

Then I suggest exposing `max-width-chars` as a property in `MessageReply` (please use the new properties macro) and then set this value to both the sender and message label.

Reply:
> Then I suggest exposing max-width-chars as a property in MessageReply

Is it really necessary? It's only set once and only used by MessageSticker for now.
My idea is to expose GtkLabel property instead of duplacating the same property in MessageReply.

Reply:
I think this is an optimal solution
```rust 
pub(crate) fn set_max_char_width(&self, n_chars: i32) {
        self.imp().message_label.set_max_width_chars(n_chars);
        self.imp().sender_label.set_max_width_chars(n_chars);
    }
```

Reply:
> duplacating the same property in MessageReply.

You can create a property without storing it inside the object, just call `set_max_width_chars`  to your labels in the setter.

Reply:
I want a property because this way I can set this number in the UI file.

Reply:
> I want a property because this way I can set this number in the UI file.

We are adding `MessageReply` as a child of `AdwBin` so it won't be defined in the template anyway, but maybe I don't see the whole picture